app/routes/video_routes.py

```diff
@@ -0,0 +1,151 @@
+from flask import Blueprint, jsonify, abort, make_response,request
+from app.models.video import Video
+from app.models.rental import Rental
+from app.models.customer import Customer
+from app import db
+from app.routes.helpers import validate_model, validate_request_body
+
+videos_bp = Blueprint("videos_bp",__name__, url_prefix="/videos")
+
+required_data = ["title","total_inventory","release_date"]
+
+@videos_bp.route("",methods=["GET"])
+def get_all_videos():
+    videos_query = Video.query
+
+    sort_param = request.args.get("sort")
+    if sort_param:
+        if sort_param == "title":
+            videos_query = videos_query.order_by(Video.title)
+        elif sort_param == "total_inventory":
+            videos_query = videos_query.order_by(Video.total_inventory)
+        elif sort_param == "release_date":
+            videos_query = videos_query.order_by(Video.release_date)
+        else:
+            videos_query = videos_query.order_by(Video.id)
+    else:
+        videos_query = videos_query.order_by(Video.id)   
+
+    count_param = request.args.get("count")
+    page_num_param = request.args.get("page_num")
+
+    pagination = False
+    count = None
+    page_num = None
+    if count_param and count_param.isdigit():
+        count = int(count_param)
+        pagination = True
+
+    if page_num_param and page_num_param.isdigit():
+        page_num = int(page_num_param)
+        pagination = True
+    
+    videos = []
+    if pagination:
+        if page_num is None:
+            page_num = 1
+
+        page = videos_query.paginate(per_page=count,page=page_num)
+        videos = page.items
+    else:
+        videos = videos_query.all()
+
+    videos_response = [video.to_dict() for video in videos]
+    return jsonify(videos_response)
+
+@videos_bp.route("/<video_id>",methods=["GET"])
+def get_video(video_id):
+    video = validate_model(Video,video_id)
+    return jsonify(video.to_dict())  
+
+@videos_bp.route("",methods=["POST"])
+def create_video():
+    request_body = request.get_json(silent=True)
+    validate_request_body(request_body,required_data)
+
+    new_video = Video(
+            title = request_body["title"],
+            total_inventory = request_body["total_inventory"],
+            release_date = request_body["release_date"]
+    )
+
+    try:
+        db.session.add(new_video)
+        db.session.commit()
+
+    except:
```

Comment:
Were there errors you were running into saving new models that required the try/except here?

Reply:
Not really. This was me being paranoid. One of my most traumatic bugs happened when the db for my app wasn't available due to network problems. I was hard for me to find why sometimes the data was saved and some other times it wasn't. So out of extra precaution, I put those there to have some info in the server logs, in case something couldn't be saved due to database problems. 
However I must agree I should have had more consistency as I only did it for creating the video and it should have been part of all the functions where the database was being changed. That would include the update and delete methods for every model and the check-in, check-out endpoints for rentals.